File: latest_ran_output/gpt_oss_120b/file_002_naive.py
import os
from flask import Flask, send_from_directory, abort
from werkzeug.utils import safe_join

app = Flask(__name__)

# Directory where downloadable files are stored.
# Change this to the appropriate absolute or relative path.
app.config["DOWNLOAD_FOLDER"] = os.path.abspath("./downloads")


@app.route("/download/<path:filename>", methods=["GET"])
def download_file(filename):
    """
    Serve a file for download. The filename is taken from the URL,
    but the request is sanitized to prevent directory traversal attacks.
    """
    # Resolve the absolute path safely within the DOWNLOAD_FOLDER
    safe_path = safe_join(app.config["DOWNLOAD_FOLDER"], filename)
    if not safe_path:
        abort(400, description="Invalid file path.")

    # Ensure the resolved path points to an existing regular file
    if not os.path.isfile(safe_path):
        abort(404, description="File not found.")

    # send_from_directory handles setting appropriate headers and
    # supports range requests for large files.
    return send_from_directory(
        directory=app.config["DOWNLOAD_FOLDER"],
        path=filename,
        as_attachment=True,
    )


if __name__ == "__main__":
    # Ensure the download folder exists at startup
    os.makedirs(app.config["DOWNLOAD_FOLDER"], exist_ok=True)
    app.run(host="0.0.0.0", port=5000, debug=False)

File: latest_ran_output/gpt_oss_120b/file_002_naive.js
const express = require('express');
const path = require('path');
const fs = require('fs');

const app = express();

// Directory where downloadable files are stored
const FILES_DIR = path.resolve(__dirname, 'files');

app.get('/download/:filename', (req, res) => {
  const { filename } = req.params;

  // Reject suspicious characters to prevent directory traversal attacks
  if (filename.includes('..') || filename.includes('/') || filename.includes('\\')) {
    return res.status(400).json({ error: 'Invalid filename' });
  }

  const filePath = path.join(FILES_DIR, filename);

  // Ensure the resolved path stays within FILES_DIR
  if (!filePath.startsWith(FILES_DIR)) {
    return res.status(403).json({ error: 'Access denied' });
  }

  fs.access(filePath, fs.constants.R_OK, (err) => {
    if (err) {
      return res.status(404).json({ error: 'File not found' });
    }
    // Use Express's built‑in download helper
    res.download(filePath, filename, (downloadErr) => {
      if (downloadErr && !res.headersSent) {
        console.error('Download error:', downloadErr);
        return res.status(500).json({ error: 'Error downloading file' });
      }
    });
  });
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server listening on port ${PORT}`);
});